### PR TITLE
Implement auth.teams.list api method

### PR DESCRIPTION
**Pull request by DavyJohnes**
##### Pull Request Guidelines

These are recommendations for pull requests.
They are strictly guidelines to help manage expectations.

##### PR preparation
Run `make pr-prep` from the root of the repository to run formatting, linting and tests.

##### Should this be an issue instead
- [ ] is it a convenience method? (no new functionality, streamlines some use case)
- [ ] exposes a previously private type, const, method, etc.
- [ ] is it application specific (caching, retry logic, rate limiting, etc)
- [ ] is it performance related.

##### API changes

Since API changes have to be maintained they undergo a more detailed review and are more likely to require changes.

- no tests, if you're adding to the API include at least a single test of the happy case.
- If you can accomplish your goal without changing the API, then do so.
- dependency changes. updates are okay. adding/removing need justification.

###### Examples of API changes that do not meet guidelines:
- in library cache for users. caches are use case specific.
- Convenience methods for Sending Messages, update, post, ephemeral, etc. consider opening an issue instead.

### CODE REVIEW

**Comment by zchee** [comment]
@DavyJohnes Sorry for late, seem to LGTM. will review soon.

File: users.go [hunk 1]
```diff
@@ -292,6 +292,12 @@ func GetUsersOptionPresence(n bool) GetUsersOption {
 	}
 }
 
+func GetUsersOptionTeamID(t string) GetUsersOption {
+	return func(p *UserPagination) {
+		p.teamID = t
+	}
+}
+
```
Comment (kanata2) [inline users.go:300]: unused?

File: team.go [hunk 2]
```diff
@@ -165,3 +185,21 @@ func (api *Client) GetBillableInfoForTeamContext(ctx context.Context) (map[strin
 
 	return api.billableInfoRequest(ctx, "team.billableInfo", values)
 }
+
+func (api *Client) GetAuthTeams() ([]TeamInfoShort, error) {
+	return api.GetAuthTeamsContext(context.Background())
+}
```
Comment (kanata2) [inline team.go:191]: `ListAuthTeams` looks better 😄

File: team.go [hunk 3]
```diff
@@ -95,6 +105,16 @@ func (api *Client) accessLogsRequest(ctx context.Context, path string, values ur
 	return response, response.Err()
 }
 
+func (api *Client) authTeamsRequest(ctx context.Context, path string, values url.Values) (*AuthTeamsResponse, error) {
```
Comment (kanata2) [inline team.go:108]: This method can be added when `auth.teams` specific processing is needed.

File: team.go [hunk 4]
```diff
@@ -165,3 +185,21 @@ func (api *Client) GetBillableInfoForTeamContext(ctx context.Context) (map[strin
 
 	return api.billableInfoRequest(ctx, "team.billableInfo", values)
 }
+
+func (api *Client) GetAuthTeams() ([]TeamInfoShort, error) {
+	return api.GetAuthTeamsContext(context.Background())
+}
+
+func (api *Client) GetAuthTeamsContext(ctx context.Context) ([]TeamInfoShort, error) {
```
Comment (kanata2) [inline team.go:193]: Could you support `include_icon` param?

https://api.slack.com/methods/auth.teams.list

**Review by kanata2 — CHANGES_REQUESTED** [review]
Thanks!
I've commented. Please confirm 🙏

**Pull request closed without merging**